### PR TITLE
[GH-24] Fix panic when user == nil

### LINKED ISSUE #24: Handle panic and crash when user is returned as nill from Azure
With 1 particular user, I am seeing the authentication  process  return a 500 and the plugin will crash. 
Other user are not affected. 

In any case, we should handle this to prevent the crash should anyone else come accross a case like this where something seems to be invalid with the users. 

See logs
```
{"level":"debug","ts":1603489636.774614,"caller":"web/handlers.go:100","msg":"Received HTTP request","method":"POST","url":"/api/v4/logs","request_id":"ho6mugii7bb5jbct4367eni1ge","status_code":"200"}
{"level":"debug","ts":1603489643.640474,"caller":"wsapi/websocket_handler.go:25","msg":"Websocket request","action":"user_update_active_status"}
{"level":"debug","ts":1603489647.0510972,"caller":"plugin/hclog_adapter.go:53","msg":"panic: runtime error: invalid memory address or nil pointer dereference","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.05122,"caller":"plugin/hclog_adapter.go:53","msg":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e792b6]","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.051265,"caller":"plugin/hclog_adapter.go:53","msg":"","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.0513,"caller":"plugin/hclog_adapter.go:53","msg":"goroutine 6370 [running]:","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.0513449,"caller":"plugin/hclog_adapter.go:53","msg":"main.(*Plugin).completeUserOAuth(0xc000387900, 0x2620680, 0xc0005f9650, 0xc000294300)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.051382,"caller":"plugin/hclog_adapter.go:53","msg":"\t/Users/dylanhaussermann/go/src/github.com/mattermost/mattermost-plugin-msteams-meetings/server/http.go:137 +0x7d6","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1603489647.0514162,"caller":"plugin/hclog_adapter.go:53","msg":"main.(*Plugin).ServeHTTP(0xc000387900, 0xc00007ae40, 0x2620680, 0xc0005f9650, 0xc000294300)","plugin_id":"com.mattermost.msteamsmeetings"}
```

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Check whether user == nil when getting the user.

#### Ticket Link
Fix #24

### CODE REVIEW

File: server/authorization.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func (p *Plugin) authenticateAndFetchUser(userID, userEmail, channelID string) (
 		return nil, &authError{Message: oauthMsg, Err: apiErr}
 	}
 	user, err = p.getUserWithToken(userInfo.OAuthToken)
-	if err != nil || user == nil {
+	if err != nil {
 		return nil, &authError{Message: oauthMsg, Err: apiErr}
```
Comment (mickmister) [inline server/authorization.go:46]:
```suggestion
		return nil, &authError{Message: oauthMsg, Err: err}
```

File: server/authorization.go [hunk 2]
```diff
@@ -42,7 +42,7 @@ func (p *Plugin) authenticateAndFetchUser(userID, userEmail, channelID string) (
...
```
Comment (mickmister) [inline server/authorization.go:42]: Do we want to watch out for the same issue here?

**Review by mickmister — CHANGES_REQUESTED** [review]
Change looks good, just one request to fix an existing issue

File: server/authorization.go (hunk 2 above)
Comment (larkox) [reply server/authorization.go:42]: With the current usage, Err will be nil, but that will not be any problem, since authError will be a non-nil value.

Comment (mickmister) [reply server/authorization.go:42]: So if `apiErr` is nil, then we don't have anything to log right? Just so I understand, we do need to return and log an error (`authError.Err`) when we fail to fetch the graph user, but not when the user isn't found in the store.

**Review by mickmister — APPROVED** [review]

**Review by iomodo — APPROVED** [review]
LGTM!

**Comment by DHaussermann** [comment]
Hi @larkox 
With the affected user. I still see a panic when trying to connect. I have `f00ca1c` deployed which includes a merge from master. 

The affected user see's a 500 in the new tab that opens and logs show:
```
{"level":"warn","ts":1605194243.70717,"caller":"cluster/cluster.go:631","msg":"Cluster is not initialized"}
{"level":"debug","ts":1605194243.70728,"caller":"web/handlers.go:100","msg":"Received HTTP request","method":"POST","url":"/api/v4/channels/members/me/view","request_id":"4hsauh97kjbxzbkgzik5m5j1bw","status_code":"200"}
{"level":"debug","ts":1605194244.7280068,"caller":"plugin/hclog_adapter.go:53","msg":"panic: runtime error: invalid memory address or nil pointer dereference","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728099,"caller":"plugin/hclog_adapter.go:53","msg":"[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1e793e6]","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728127,"caller":"plugin/hclog_adapter.go:53","msg":"","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.7281432,"caller":"plugin/hclog_adapter.go:53","msg":"goroutine 84 [running]:","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728157,"caller":"plugin/hclog_adapter.go:53","msg":"main.(*Plugin).completeUserOAuth(0xc000293780, 0x26207e0, 0xc00051fcc0, 0xc000268100)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728173,"caller":"plugin/hclog_adapter.go:53","msg":"\t/Users/dylanhaussermann/go/src/github.com/mattermost/mattermost-plugin-msteams-meetings/server/http.go:137 +0x7d6","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.7281878,"caller":"plugin/hclog_adapter.go:53","msg":"main.(*Plugin).ServeHTTP(0xc000293780, 0xc00049ad20, 0x26207e0, 0xc00051fcc0, 0xc000268100)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728218,"caller":"plugin/hclog_adapter.go:53","msg":"\t/Users/dylanhaussermann/go/src/github.com/mattermost/mattermost-plugin-msteams-meetings/server/http.go:39 +0x167","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.7282581,"caller":"plugin/hclog_adapter.go:53","msg":"github.com/mattermost/mattermost-server/v5/plugin.(*hooksRPCServer).ServeHTTP(0xc00028f640, 0xc00028fda0, 0x355a348, 0x0, 0x0)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728322,"caller":"plugin/hclog_adapter.go:53","msg":"\t/Users/dylanhaussermann/go/pkg/mod/github.com/mattermost/mattermost-server/v5@v5.28.1/plugin/client_rpc.go:365 +0x29b","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.72835,"caller":"plugin/hclog_adapter.go:53","msg":"reflect.Value.call(0xc000410840, 0xc000418188, 0x13, 0x2477e6d, 0x4, 0xc000402f08, 0x3, 0x3, 0x1f87ee0, 0xc00048d800, ...)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728367,"caller":"plugin/hclog_adapter.go:53","msg":"\t/usr/local/go/src/reflect/value.go:460 +0x8ab","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728392,"caller":"plugin/hclog_adapter.go:53","msg":"reflect.Value.Call(0xc000410840, 0xc000418188, 0x13, 0xc0000d3708, 0x3, 0x3, 0x0, 0x0, 0x0)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.72841,"caller":"plugin/hclog_adapter.go:53","msg":"\t/usr/local/go/src/reflect/value.go:321 +0xb4","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728438,"caller":"plugin/hclog_adapter.go:53","msg":"net/rpc.(*service).call(0xc000422000, 0xc0004160f0, 0xc00041a090, 0xc00041a0a0, 0xc00042a580, 0xc0004005c0, 0x1f67860, 0xc00028fda0, 0x16, 0x1f8a7a0, ...)","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728485,"caller":"plugin/hclog_adapter.go:53","msg":"\t/usr/local/go/src/net/rpc/server.go:377 +0x17f","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728511,"caller":"plugin/hclog_adapter.go:53","msg":"created by net/rpc.(*Server).ServeCodec","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"debug","ts":1605194244.728524,"caller":"plugin/hclog_adapter.go:53","msg":"\t/usr/local/go/src/net/rpc/server.go:474 +0x42b","plugin_id":"com.mattermost.msteamsmeetings"}
{"level":"error","ts":1605194244.7312052,"caller":"mlog/log.go:229","msg":"Plugin failed to ServeHTTP, RPC call failed","plugin_id":"com.mattermost.msteamsmeetings","error":"unexpected EOF"}
{"level":"debug","ts":1605194244.731414,"caller":"plugin/hclog_adapter.go:51","msg":"plugin process exited","plugin_id":"com.mattermost.msteamsmeetings","wrapped_extras":"pathplugins/com.mattermost.msteamsmeetings/server/dist/plugin-darwin-amd64pid51951errorexit status 2"}
{"level":"debug","ts":1605194253.254185,"caller":"web/handlers.go:100","msg":"Received HTTP request","method":"POST","url":"/api/v4/users/status/ids","request_id":"u7jxxqpa5jrw8dnut4x1b8nb4a","status_code":"200"}
```

**Comment by DHaussermann** [comment]
@larkox I can test this but. I now see a lint error 
```
Running golangci-lint
golangci-lint run ./...
server/http.go:136:61: `propertly` is a misspelling of `property` (misspell)
		http.Error(w, "User has no mail. Please check the user is propertly configured in Microsoft", http.StatusInternalServerError)
		                                                          ^
server/http.go:142:59: `propertly` is a misspelling of `property` (misspell)
		http.Error(w, "User has no ID. Please check the user is propertly configured in Microsoft", http.StatusInternalServerError)
		                                                        ^
server/http.go:148:76: `propertly` is a misspelling of `property` (misspell)
		http.Error(w, "User has no user principal name. Please check the user is propertly configured in Microsoft", http.StatusInternalServerError)
```

**Comment by DHaussermann** [comment]
Tested to ensure panic is resolved. 
- The browser now shows me the real problem with the user rather than a 500. 
- Logging also shows this error and no panic or crash

I see `User has no mail. Please check the user is propertly configured in Microsoft`

Leaving this open for @larkox  to review:
- Lint error above
- Do we have control over this error text? The word `properly` is spelt wrong as `propertly`

**Pull request merged**